### ISSUE TITLE
Spamming requests to {motd,update}.etlegacy.com

### ISSUE THREAD

**Issue by ETLegacyTeam** [etlegacy/etlegacy#666]
Author: @saukko28 (Saukko)
Date: 2014-11-27
Redmine Issue: 666, https://dev.etlegacy.com/issues/666

---

Hello all, this is far from worth of being the ticket from Hell but this gets spammed in the console by following the next steps:
1. Start etl.exe
2. Start pressing ESC continously like 20 times f.e.
3. Check console

This was just an example but this works also in a bit more of a normal way:
1. Go to 'Options'
2. Go back by pressing 'Back' or press ESC
3. Check console

This message comes only once each time you go back to the main menu. So here's an example how you get the mssage three times for instance:
1. Go to 'Play Online'
2. Go back
3. Go to 'Options'
4. Go back
5. Go to 'Mods'
6. Go back

It's not a big of a deal but I think it's something to be considered.

-*S

---

Related issue(s): #32

---

Related commit(s):
3e52d5e65de823e17c6989ba6efaa9f8b29a97de - general: separated motd check from update check, r..
 46e62c484fdae4edec656620e6827223eab3f2a0 - Revert "general: separated motd check from update ..
 68e019632c72eaa5209f2cfedab42acff6e2f218 - the client will check for updates every 5 minutes,..

**Comment by ETLegacyTeam** [comment]
From: @jackeri (Jacker)
Date: 2014-12-18T08:42:01Z

---

While someone might consider this spam, to me it is not that. If the client does not check data from the master server often it can miss out on important information such as updates (this can happen while the user has the client open). Besides the data sent/received is very minuscule and does not really affect anything that much.

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2014-12-18T14:47:49Z

---

Jacker wrote:
> While someone might consider this spam, to me it is not that. If the client does not check data from the master server often it can miss out on important information such as updates (this can happen while the user has the client open). Besides the data sent/received is very minuscule and does not really affect anything that much.

True. Vanilla client has been behaving this way for years. I however still believe it's overkill: update and motd are checked every time the player access the main menu.

On my previous commit, I tried to separate motd checks from update checks, by checking motd only when the player opens the "Play online" window. This didn't solve the "check update every time", but lessens the "motd spam". This has been reverted since vanilla client didn't really like that change (all menu strings disappearing).

As a very cheap workaround, we could hide the "motd" print in console for the time being.
